src/redux/reducer.js

```diff
@@ -0,0 +1,78 @@
+import {combineReducers} from 'redux'
+import {
+  CREATE_MOVIE,
+  DELETE_MOVIE,
+  EDIT_MOVIE,
+  Filter,
+  FILTER_MOVIES, SET_MOVIES,
+  Sort,
+  SORT_MOVIES,
+  UPDATE_MOVIE
+} from './actions'
+
+function moviesAPI(state = [], action) {
+  switch (action.type) {
+    case SET_MOVIES:
+      return {
+        ...state,
+        movies: action.movies
+      };
+    case SORT_MOVIES:
+      const key = action.payload;
+      return {
+        movies: state.movies.sort((a, b) => a[key] > b[key] ? 1 : -1)
```

Comment:
should also return `...state`
in your case it just overrides the store with only one `movies` item